include/seqan3/range/view/pairwise_combine.hpp

```diff
@@ -0,0 +1,307 @@
+     * the original begin and the end of the underlying range and maintains a pair of iterators which iterate over
+     * the same underlying range. The end is reached, when the second iterator points to the end of the underlying range
+     * and the first iterator to the last element of the range.
+     */
+    template <std::ForwardIterator source_iterator_type>
+    class iterator_type
+    {
+        //!\brief Alias for the value type of the underlying iterator type.
+        using underlying_val_t = typename std::iterator_traits<source_iterator_type>::value_type;
+        //!\brief Alias for the reference type of the underlying iterator type.
+        using underlying_ref_t = typename std::iterator_traits<source_iterator_type>::reference;
+    public:
+        /*!\name Associated types
+         * \{
+         */
+        using difference_type       = std::ptrdiff_t;
+        using value_type            = std::tuple<underlying_val_t, underlying_val_t>;
+        using reference             = std::tuple<underlying_ref_t, underlying_ref_t>;
+        using pointer               = void;
+        using iterator_category     = typename std::iterator_traits<source_iterator_type>::iterator_category;
+        //!\}
+
+        /*!\name Constructors, destructor and assignment
+         * \{
+         */
+        constexpr iterator_type() = default;
+        constexpr iterator_type(iterator_type const &) = default;
+        constexpr iterator_type(iterator_type &&) = default;
+        constexpr iterator_type & operator=(iterator_type const &) = default;
+        constexpr iterator_type & operator=(iterator_type &&) = default;
+        ~iterator_type() = default;
+
+        /*!\brief Constructs the iterator from the iterator pointing to the last element and the iterator to the first
+         *        position.
+         * \param[in] i The iterator pointing to the first element of the underlying range.
+         * \param[in] e The iterator pointing to end of the underlying range.
+         *
+         * \details
+         *
+         * Constructs the iterator by caching the end of the underlying range and setting the first iterator to the
+         * given position and the second to the first incremented by one.
+         */
+        constexpr iterator_type(source_iterator_type i,
+                                source_iterator_type e) noexcept  : first_it{i},
+                                                                        second_it{++i},
+                                                                        end_it{e}
+        {}
+        //!\}
+
+        /*!\name Accessors
+         * \{
+         */
+         //!\brief Accesses the pointed-to element
+        constexpr reference operator*() const
+            noexcept(noexcept(*std::declval<source_iterator_type>()))
+        {
+            return {*first_it, *second_it};
+        }
+        //!\}
```

Comment:
No we cannot, since we are generating temporary tuples. The operator-> returns a pointer to the pointed-to element but in this case we cannot take the address of the temporary

Reply:
Maybe we should document then that the iterator does not model the LegacyForwardIterator from the stl but the ranges::ForwardIterator? I just had a similar case

Reply:
Since we do not require LegacyIterators I don't see why this is necessary.

Reply:
@h-2 ? You just made me document it so we should decide on one or the other

Reply:
🤦‍♂️ I was just irritated because I was looking at the operator->, since it won't be used inside of stl::algorithms anyway. But now I remember, that the Cpp17ForwardIterator requires that the operator* returns a reference to the value type, which is in fact not guaranteed. So my first impression was that the users should not be bugged with even more concepts/requirements and dependencies that they have to understand, but this has far-reaching implications.